HeimdallrTests/Core/HeimdallrSpec.swift

```diff
@@ -21,6 +21,41 @@ class OAuthAccessTokenMockStore: OAuthAccessTokenStore {
     }
 }
 
+class OAuthAccessTokenInterceptorParser: OAuthAccessTokenParser {
+    
+    private let defaultParser = OAuthAccessTokenDefaultParser()
+    
+    var parseAccessTokenCalled: Bool = false
+    
+    private var shouldIntercept: Bool = false
+    private var interceptToken: OAuthAccessToken?
+    
+    func intercept(parsedToken token: OAuthAccessToken?) {
+        shouldIntercept = true
+        interceptToken = token
+    }
+    
+    func reset() {
+        parseAccessTokenCalled = false
+        shouldIntercept = false
+    }
+    
+    func parse(data: NSData) -> ParsedToken {
```

Comment:
Yeah, good point! I think since we are already exposing `Result` with our `authenticateRequest` function it might be ok to use it here as well? I think i would prefer it over the `ParsedToken` type. What do you think?